openrelik_worker_common/mount_utils.py

```diff
@@ -0,0 +1,266 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+import subprocess
+from uuid import uuid4
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class BlockDevice:
+    """BlockDevice provides functionality to map image files to block devices
+    and mount them.
+
+    Usage:
+        bd = BlockDevice('/folder/path_to_disk_image.dd')
+        mountpoints = bd.mount()
+        # Do the things you need to do :)
+        bd.destroy()
+    """
+
+    def __init__(self, image_path: str):
+        """Initialize BlockDevice class instance.
+
+        Args:
+            image_path (str): path to the image file to map and mount.
+        """
+        self.image_path = image_path
+        self.blkdevice = None
+        self.blkdeviceinfo = None
+        self.partitions = []
+        self.mountpoints = []
+        self.mountroot = "/mnt"
+
+        # Setup the loop device
+        self._losetup()
+
+        # Parse block device info
+        self._blkinfo()
+
+        # Parse partition information
+        self._parse_partitions()
+
+    def _losetup(self):
+        """Map image file to loopback device using losetup.
+
+        Returns: None
+        """
+        losetup_command = [
+            "sudo",
```

Comment:
Do we need sudo here? The worker is running as root, and if not we need to make sure that the user can sudo without password.

Reply:
Got it. Thanks for the explanation.